### PR TITLE
Rewrite panning

### PR DESCRIPTION
**Pull request by mixcas**
Now it works based on the angle and distance from the center of the window.
- add a bunch of comments
- add speed parameter
- add custom events for start/stop panning

### CODE REVIEW

File: src/js/main.js [hunk 1]
```diff
@@ -168,94 +170,145 @@ Site.Map = {
...
-      return true;
+      return false;
     }
 
-    // Get current map position
+    // Get the angle between center and mouse position
+    var angle = _this.angleFromCenter();
+
+    // Get distance between center and mouse position
+    var distance = _this.distanceFromCenter() * _this.panSpeed;
```
Comment (varffs) [inline src/js/main.js]: the comment doesnt seem to follow the code. this gets that distance and manipulates it into a distance of travel?

File: src/js/main.js [hunk 2]
```diff
@@ -168,94 +170,145 @@ Site.Map = {
...
+    // If we don't know the map's current postion we get it
     if (!_this.mapPosition) {
       _this.mapPosition = _this.getMapPosition();
     }
 
-    // Move up
-    if (_this.pan.up && _this.mapPosition[5] <= 0) {
-      _this.mapPosition[5] = _this.mapPosition[5] + 1;
+    // Get new coordinates based on the angle and the distance
+    var newX = _this.mapPosition[4] + (distance * Math.cos(angle));
```
Comment (varffs) [inline src/js/main.js]: `.cos` is the curves i meant :)

File: src/js/main.js [hunk 3]
```diff
@@ -168,94 +170,145 @@ Site.Map = {
...
+    // Check for left limit
+    if (newX >= 0) {
+      newX = 0;
     }
 
-    // Move down
-    if (_this.pan.down && _this.mapPosition[5] >= (_this.window.height * -2)) {
-      _this.mapPosition[5] = _this.mapPosition[5] - 1;
+    // Check for right limit
+    if (newX <= _this.window.width * -2) {
```
Comment (varffs) [inline src/js/main.js:297]: why negative values?

**Review by varffs — APPROVED** [review]
Works well. Makes enough sense in the code just 2 questions

File: src/js/main.js (hunk 1 above)
Comment (mixcas) [reply src/js/main.js]: True, will fix

File: src/js/main.js (hunk 3 above)
Comment (mixcas) [reply src/js/main.js:297]: That's how transform works. The transform only happens in negative values in this case. `transform: translate(0,0)` means the default position, top-left aligned. Our range of movement goes from 0 to -2windowWitdth for x-axis and from 0 to -2windowHeight.

File: src/js/main.js [hunk 4]
```diff
@@ -170,14 +170,15 @@ Site.Map = {
   handleMouseMove: function(event) {
     var _this =  this;
 
-    // Save mouse position
-    _this.mouse.x = event.clientX;
-    _this.mouse.y = event.clientY;
+      // Save mouse position
+      _this.mouse.x = event.clientX;
```
Comment (varffs) [inline src/js/main.js]: ?

File: src/js/main.js [hunk 5]
```diff
@@ -212,7 +217,46 @@ Site.Map = {
...
+    _this.startPanEvent();
+
     window.requestAnimationFrame(_this.pan.bind(_this));
+
+  },
+
+  startPanEvent: function() {
+    var _this =  this;
+
+    var startPanEvent = new CustomEvent('startpan', {
```
Comment (varffs) [inline src/js/main.js]: doesnt actually work in IE (https://caniuse.com/#search=CustomEvent) but I guess we don't really care

**Pull request merged**